zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/authorization/AuthorizationStateTest.java

```diff
@@ -249,4 +249,34 @@ void shouldStorePermissionsByPermissionType() {
     // then
     assertThat(resourceIds1).isNotEqualTo(resourceIds2);
   }
+
+  @Test
+  void shouldInsertOwnerTypeByKey() {
+    // given
+    final var ownerKey = 1L;
+    final var ownerType = AuthorizationOwnerType.USER;
+
+    // when
+    authorizationState.insertOwnerTypeByKey(ownerKey, ownerType);
+
+    // then
+    final var persistedOwnerType = authorizationState.getOwnerType(ownerKey);
+    assertThat(persistedOwnerType).contains(ownerType);
+  }
+
+  @Test
+  void shouldNotInsertOwnerTypeByKeyTwice() {
```

Comment:
❓ Is this a possible scenario ?

Reply:
Realistically not, but as there will be multiple different events adding owners to this state I think it's still a valuable test to have. Better safe than sorry 🤷

Reply:
Should we reject the command if this happens ? Or do you think it's not a problem because the owner type is already there

Reply:
I don't feel the need to explicitly reject the command for this. It means checking the state in the processor for a case that can't really happen. If it does happen it indicates a bigger problem as we'd have a state inconsistency.
What happens now is it raises an exception which also results in a command rejection, except it will be visible in our logging as a reported error.

Reply:
Got it, thanks for the explanation